### PR TITLE
Update getBlock/getTransaction rpc handling of return_data

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
`simulateTransaction` was reworked to use base64 encoding for return_data here: https://github.com/solana-labs/solana/pull/24355 But getBlock and getTransaction are still returning an array of u8s, plus a snake-case `program_id` field label.

#### Summary of Changes
Move `RpcTransactionReturnData` to `solana_transaction_status::UiTransactionReturnData` to use in `UiTransactionStatusMeta`, and re-export in `solana_rpc_client_api`
Also set `return_data` to skip serializing if None

This is a breaking change for consumers of `UiTransactionStatusMeta::return_data`, but that field has not been released as stable yet.

### CODE REVIEW

File: rpc-client-api/src/response.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+pub use solana_transaction_status::{
+    UiReturnDataEncoding as ReturnDataEncoding, UiTransactionReturnData as RpcTransactionReturnData,
+};
```
Comment (joncinque) [inline rpc-client-api/src/response.rs:3]: Considering this is a breaking change, how about breaking the current consumers loudly on build by not doing these re-exports? Would that still work with the backwards compatibility policy for unstable APIs?

**Review by joncinque — DISMISSED** [review]
Thanks for fixing this! Just the one question, but nothing that should stop this from going forward

File: docs/src/developing/clients/jsonrpc-api.md [hunk 2]
```diff
@@ -443,6 +443,9 @@ The result field will be an object with the following fields:
       - `loadedAddresses: <object|undefined>` - Transaction addresses loaded from address lookup tables. Undefined if `maxSupportedTransactionVersion` is not set in request params.
         - `writable: <array[string]>` - Ordered list of base-58 encoded addresses for writable loaded accounts
         - `readonly: <array[string]>` - Ordered list of base-58 encoded addresses for readonly loaded accounts
+      - `returnData: <object|undefined` - Latest return data produced by an instruction or inner instruction in the transaction:
```
Comment (joncinque) [inline docs/src/developing/clients/jsonrpc-api.md]:
```suggestion
      - `returnData: <object|undefined>` - Latest return data produced by an instruction or inner instruction in the transaction:
```

File: docs/src/developing/clients/jsonrpc-api.md [hunk 3]
```diff
@@ -443,6 +443,9 @@ The result field will be an object with the following fields:
...
+        - `programId: <string>` - the program that generated the data
```
Comment (joncinque) [inline docs/src/developing/clients/jsonrpc-api.md]:
```suggestion
        - `programId: <string>` - the program that generated the data, as base-58 encoded string
```

File: docs/src/developing/clients/jsonrpc-api.md [hunk 4]
```diff
@@ -3045,6 +3048,9 @@ Returns transaction details for a confirmed transaction
     - `loadedAddresses: <object|undefined>` - Transaction addresses loaded from address lookup tables. Undefined if `maxSupportedTransactionVersion` is not set in request params.
       - `writable: <array[string]>` - Ordered list of base-58 encoded addresses for writable loaded accounts
       - `readonly: <array[string]>` - Ordered list of base-58 encoded addresses for readonly loaded accounts
+    - `returnData: <object|undefined` - Latest return data produced by an instruction or inner instruction in the transaction:
```
Comment (joncinque) [inline docs/src/developing/clients/jsonrpc-api.md]:
```suggestion
    - `returnData: <object|undefined>` - Latest return data produced by an instruction or inner instruction in the transaction:
```

File: docs/src/developing/clients/jsonrpc-api.md [hunk 5]
```diff
@@ -3045,6 +3048,9 @@ Returns transaction details for a confirmed transaction
...
+      - `programId: <string>` - the program that generated the data
```
Comment (joncinque) [inline docs/src/developing/clients/jsonrpc-api.md]:
```suggestion
      - `programId: <string>` - the program that generated the data, as base-58 encoded string
```

**Review by joncinque — COMMENTED** [review]
Just a few little nits for consistency with the rest of the docs, looks great!

File: rpc-client-api/src/response.rs (hunk 1 above)
Comment (CriesofCarrots) [reply rpc-client-api/src/response.rs:3]: This'll just break consumers of return data from simulate. Yeah, works for me.

**Review by joncinque — APPROVED** [review]

**Pull request merged**